tfhe/src/integer/server_key/radix/slice.rs

```diff
@@ -0,0 +1,549 @@
+use std::ops::{Bound, RangeBounds};
+
+use crate::error::InvalidRangeError;
+use crate::integer::{RadixCiphertext, ServerKey};
+use crate::prelude::{CastFrom, CastInto};
+use crate::shortint;
+
+/// Normalize a rust bound object, and check that it is valid for the source integer
+pub(crate) fn parse_bounds<B, R>(
```

Comment:
Here I would return either a `Range` if the bound object is normalized to a exclusive range (caller then has to use .start(), .end()), or a `RangeInclusive` if the bound object is normalized to an inlusive range (and caller then uses .into_inner())

Reply:
good idea !

Reply:
done